fdbserver/ptxn/MutableTeamPeekCursor.actor.h

```diff
@@ -97,8 +97,12 @@ class MutableTeamPeekCursor : public BaseClass {
 public:
 	MutableTeamPeekCursor(const UID serverID_,
 	                      const StorageTeamID& privateMutationStorageTeamID_,
-	                      const TLogInterfaceByStorageTeamIDFunc& getTLogInterfaceByStorageTeamID_)
-	  : BaseClass(), serverID(serverID_), storageServerToTeamIDKey(::storageServerToTeamIdKey(serverID_)),
+	                      const TLogInterfaceByStorageTeamIDFunc& getTLogInterfaceByStorageTeamID_,
+	                      const Version& initialVersion_ = 0)
+
+	  : // This minus one in BaseClass is necessary since all new cursor will start with BaseClass::currentVersion + 1
+	    BaseClass(initialVersion_ - 1), serverID(serverID_),
```

Comment:
If we want all the cursors to take the first version it does not know, I think the approach might be a bit different. 

1. So we know the StorageTeamPeekCursor is starting with `rpcVersion`, which is the version it does not know. 

2. Also, we want the `BroadcastedStorageTeamPeekCursor` to start with the first version it does not know, so far what we do is, we will call `remoteMoreAvailable` before consuming any of the data. So the first version it would use is the `version_` in the constructor. 

3. Now consider the `MutableTeamPeekCursor`, for any cursor it creates, it would use `BaseClass::currentVersion + 1` (see MutableTeamPeekCursor.h:147). In this case, if we set the initial version of `BaseClass` to be `initialVersion_ - 1`, all cursors it starts with would use `initialVersion_`.

I did thought for some time about this, I could be wrong but I think this approach actually satisfies the requirement if we want the cursor to take the first version it does not know, i.e. let the cursor takes `lastVersion`, which is one next to the latest version it received from TLog.

Reply:
I'm not sure if I understand `remoteMoreAvailable` part. What I see right now is:

StorageTeamPeerCursor(version) : rpcVersion(version)      // this is the first version the cursor doesn't know

BroadcastedStorageTeamPeekCorsur(version) : currentVersion(`version`)   // this is the last version the cursor knows
BroadcastedStorageTeamPeekCorsur::CreateCursor(`currentVersion + 1`)

MutableTeamPeekCursor(version) : BroadcastedStorageTeamPeekCorsur(`version - 1`) // convert first unknown version to last known version.


What I propose is:

BroadcastedStorageTeamPeekCorsur(`version`) : currentVersion(`version - 1`)
BroadcastedStorageTeamPeekCorsur::CreateCursor(currentVersion + 1)

MutableTeamPeekCursor(version) : BroadcastedStorageTeamPeekCorsur(`version`)

We still set `currentVersion` to the same value, it is just that now all types of cursor take the first unknown version in its constructor.